sklearn/cluster/_bicluster.py

```diff
@@ -124,7 +129,7 @@ def fit(self, X, y=None):
 
         """
         X = self._validate_data(X, accept_sparse="csr", dtype=np.float64)
-        self._check_parameters()
+        self._check_parameters(X.shape[0])
```

Comment:
This now depends on child classes to define a `_check_parameters` that accepts `n_samples`, even tho the `BaseSpectral._check_parameters` does not.

May we update `BaseSpectral._check_parameters` to accept `n_samples` to be consistent?